\chapter{Literature Review}

\section{An overview}
This thesis models \textit{Saccharomyces cerevisiae} (baker's yeast), using algebraic curves 
coupled with a nutrient partial differential equation. A review of the literature is supplied 
to motivate the modelling choices from a range of perspectives.

\section{Modelling Kingdom Fungi}

Microfossils discovered in the Grassy Bay Formation 
(Shaler Supergroup, Artic Canada) dating back to the Proterozoic Eon, $2500$ to $538.8$ million years ago, were 
found to have fungal affinity \cite{loron2019early}. Curiously, it took until 1969 for Fungi to be 
recognised in plant ecologist Robert Whittaker's work as part of a taxonomic kingdom, Kingdom Fungi,
which is distinct from Kingdom Plantae \cite{whittaker1969new}. Today, locating fungal species 
in phylogenetic lineages is facilitated through colonial morphology imaging as a first 
estimate. More advanced experimental techniques such as transmission 
electron microscopy (TEM) can provide  
conclusive identification \cite{loron2019early} of fungal samples.
\\

Morphological features observed in fungal strains such as fission yeast 
(\textit{Schizosaccharomyces pombe}) are sometimes cited as the reason 
why yeast is broadly considered a ``model organism'' \cite{hayles2018introduction}.
Cell Biologist Murdoch Mitchison was able to study the fission yeast cell cycle in 1957 for instance,
because the length of the rod shaped cells of \textit{S. pombe} reflected their stage in the cell cycle.
It happens to be an \textit{S. pombe} spore grown on malt extract agar that was imaged in exquisite detail via 
X-ray diffraction microscopy by \cite{jiang2010quantitative}. By 3D 
reconstruction techniques \cite{jiang2010quantitative} visualised the organelles
present in the unstained (and unsectioned) spore at a resolution of $50-60 \ nm$ 
producing isosurfaces of cellular organelles such as the endoplasmic reticulum, 
vacuole and cell wall.
\\

Contemporary advancements in computer hardware facilitate 
the ability for big datasets such as 3D morphological data 
obtained from yeast to be analysed mathematically. \cite{clermont2015inverse} 
discuss the ``inverse problem" in mathematical biology, which is to determine a model 
from noisy, complex and extensive biological data whilst minimising bias. This often 
comes in the framework of model parameter estimation which 
can be accelerated using fairly systematic deep learning techniques. Still, the predictive power
of a mathematical model is always limited by the bias implicit (or explicit) in its assumptions.
\\

Biology is inherently multi-scale and the problem 
of bridging between scales is named the ``translational challenge'' \cite{an2009agent}.
The interplay between the phenotypical properties of a filamentous yeast colony and 
its genotype, is one such challenge. Indeed, the eukaryote \textit{S. pombe}, has a similar 
cytoskeletal organisation to a human tissue cell as well as analogous DNA transcription mechanisms 
\cite{hoffman2015ancient} and yet human tissue 
morphology vastly differs from fission yeast.
\\

\cite{an2009agent} position agent-based models (ABMs), dating back to cellular automata, as well suited to the 
translational challenge for a number of reasons. These include the ability 
for ABMs to model morphological emergence from simple and modular rules. These rules 
are constructed from a qualitative understanding of the underlying mechanisms of for instance 
\textit{S. pombe} intracellular stresses which could be based on some basic assumptions
about cellular elasticity. Importantly, ABMs are easy to deploy acting 
as a cheap alternative to \textit{in vitro} experiments with fungal species.
\\

Multi-cellular agent based modelling of \textit{S. pombe} or
\textit{Saccharomyces cerevisiae} (baker's yeast) is of interest to a number of scientific  
communities. On the one hand, due to the molecular similarity of yeast to human 
cells, it can serve as a model organism that can ethically be grown, and on 
the other, it is of intrinsic interest to philogeneticists who study the origin 
of life on Earth. Morphological studies using ABMs can also provide breweries and 
food processing industries with practically useful findings to 
set up ideal growth conditions for cultures prior to expensive 
industrial implemenatations. Indeed \textit{pombe} is Swahili 
for ``booze'', being originally observed in contaminated 
millet beer delayed on route from East Africa to Germany \cite{hayles2018introduction}.
\\

\section{Agent based modelling in context}
In the current thesis, a dynamic off-lattice agent based modelling framework for \textit{S. cerevisiae} 
colonial morphology 
is proposed and implemented. As opposed to fission yeast, \textit{S. cerevisiae} exhibits 
budding and can be induced into a filamentous regime called pseudohyphal growth
in which cells become elongated leading to overall branching patterns 
in the colonial morphology. The off-lattice ABM developed in \cite{li2024off} 
inspired the modelling of the pseudohyphal \textit{S. cerevisiae} regime 
presented here, though it uses a vastly different computational methodology.
\\

\cite{li2024off} use approximate bayesian computatation (ABC)
to infer model parameters within the context of a probabilistic (and spatial) branching simulation, 
yielding remarkable fits to micrographs of filamentous yeast. Building 
from the experimental observations of pseudohyphal growth presented in 
\cite{gimeno1992unipolar}, Li et al. dispense with diffusion limited growth (DLG) which is
shown by \cite{tronnolone2018diffusion} to be just a peripheral mechanism in filamentous 
colony shape.
\\

Whilst pseudohyphal growth dynamics
have proved to be the dominant growth mechanism \cite{tronnolone2018diffusion} for 
\textit{S. cerevisiae} in nutrient poor environments, it is the aim of the present 
work to reintroduce a diffusive nutrient medium in the hope of defining 
\textit{to what extent} these two phenomena intertwine.
\\

\cite{brown2021rigid} develop rigid body modelling for multi-cellular colonies using 
an ABM which represents each cell as a deformable polygon. The emphasis in the model 
of Brown et al. is to represent the process of cell division in diverse biological 
settings through the addition of new nodes and edges to each polygonal cell. 
In contrast, \cite{li2024off}, use static ellipsoidal cells that are added into the simulation
abruptly. \cite{li2024off} instead focus on the ability to fit their model 
to experimental micrographs of \textit{S. cerevisiae} using statistical methods.
Interestly, this choice seems to reflect a gap in the multi-cellular modelling literature:
\textit{how can cell division be represented without the artificial inclusion of new parameters?}
\\

In both the work of \cite{van2020quantitative}, and the ABM of 
\cite{brown2021rigid}, the dynamics of the colony 
are derived from the dynamics of individual cells which are represented
by meshes with time-dependent topology as well as geometry. However ingenious this idea may be, 
computation quickly becomes intractible for large colonies
due to the increasing size of the underlying system of ordinary differential equation (ODEs)
for the node states. Most ABMs need supercomputers, however
the computationally costly use of meshed cells is circumvented in the current work by employing 
an algebraic representation of cell geometry.
\\

\section{From algorithmic botany to algorithmic mycology}

The model defined in Chapter 2 of this thesis is underpinned by a ``growing network" which is an example 
of a graph-rewriting system. The natural precursor 
to this concept is the mathematical corpus of biologist Aristid Lindenmayer, whose work 
on cellular interactions in plants and fungi based on \textit{sequential machines}, 
\cite{lindenmayer1968mathematical}, developed 
into the notion of an $L$-system, \cite{prusinkiewicz2012algorithmic}. An $L$-system
is a string-rewriting system that applies a conversion rule to each letter in a word 
simultaneously, \cite{prusinkiewicz2012algorithmic}. For example,
the rules (or ``productions'') that sends $A \rightarrow AB$ and $B \rightarrow A$,
would evolve the input word, $A$, as follows,
\begin{equation*}
\begin{split}
&A \\
&AB \\
&ABA \\
&ABAAB \\
&ABAABABA, \\
\end{split}
\end{equation*}
\\
which could continue on indefinitely. Taken together,
the alphabet, $\{A, B\}$, the productions, $\{ A \rightarrow AB, B \rightarrow A \}$, and 
the intial word, $A$, comprise an $L$-system. 
\\

The ideas  
of Lindenmayer have been taken up by the Algorthmic Botany Group \cite{algorithmicbotany_papers}
led by Przemyslaw Prusinkiewicz whose website has links to the group's papers dating
back to the 1980s. In a recent paper from 2024, Coen and Prusinkiewicz study 
two modes of developmental timing mechanisms in plants, one of them being related 
to instrinsic molecular processes and one of them being growth-dependent, for instance, dependent 
on space being available \cite{coen2024developmental}. Their computational methology,
which is of relevance to the current thesis, seems to be based on novel $L$-system
softwares developed in the Algorithmic Botany Group. Even by the year 2012, 
Boudon et al. \cite{boudon2012py} published \textit{L-Py}, a Python package 
for modelling plant architecture based on a dynamic language.  
\\

The key innovation of the present thesis work is to 
bring into contact three distinct areas of 
study into one interdependent framework for studying mycology. These three 
areas are,
\begin{itemize}
    \item Graph rewriting as exemplified in botanical $L$-system work,
    \item Constructive solid geometry (CSG) in computer vision,
    \item Reaction-diffusion partial differential equations (PDEs),
          as studied in mathematical biology \cite{turing1990chemical}.
\end{itemize}
The first and third of these areas have already been mentioned here earlier when 
referring to the work of the algorithmic botany group and the recent paper from 
\cite{tronnolone2018diffusion}. The following section addresses the second 
dot point.

\section{Implicit surfaces and constructive solid geometry (CSG)}

The three most commom ways 
of representing a 2D or 3D object mathematically are parametric, polygonal,
or implicit techniques. For instance, a parametric representation of the 
points of a circle is $(\cos{t}, \sin{t})$ where $t$
is a real number. Alternatively, a polygonal representation of a circle
would be a regular $n$-gon which could be made a better approximation 
as $n \rightarrow \infty$. This sort of approach is effectively used 
by \cite{van2020quantitative} and \cite{brown2021rigid} in their solid mechanics
frameworks. Mesh based modelling has the advantage of locality in that 
deformations to the shape can be made by moving a node in the mesh
without changing it globally which gives more flexibility.
A limitation which has already been mentioned is that mesh based models, 
particularly in the context of growth, quickly lead to huge node numbers 
which imply the necessity of supercomputers for large colony simulations.
\\

Conversely, implicit representations of smooth geometric shapes are defined 
globally at the cost of local specificity. In three dimensions, one can 
define the surface of a sphere as set of points in $\mathbb{R}^3$ 
which satisfy the following equation,
\begin{equation*}
    x^2 + y^2 + z^2 - 1 = 0.
\end{equation*}
This is a special case of what is known as a quadric surface,
which is defined as the set of points which satisfy $f(x,y,z) =0$, where
\begin{equation*}
\begin{split}
    f(x,y,z) = p_1 x^2 + p_2 xy + p_3 xz + p_4 x + p_5 y^2 + p_6 yz + p_7 y + p_8 z^2 + p_9 z + p_{10},
\end{split}
\end{equation*}
as introduced in \cite{blinn1982generalization}. The full generalisation of 
implicit curves would require a precise notion of what space 
the functions, $f$, belong to. In the case of quadric surfaces, 
the functions belong to the ring of polynomials in $x$, $y$ and $z$, written as 
$\mathbb{R}[x,y,z]$, but more exotic settings can be used.
\\

In many contexts where specificity and local shape control are essential, 
implicit curves are not the best tool. Consider the quadric surface defined above.
The only way to modify the surface is to tune the parameters $p_1, ..., p_{10}$ but 
doing this results in a change to the whole surface which requires 
advanced mathematical machinery to control. This is where constructive solid geometry (CSG) comes 
in. 
\\

CSG is a technique used to build new implicitly represented shapes from old ones.
In multi-cellular modelling, where the individual cell geometry 
is more or less fixed but there 
are regular topological bifurcations corresponding to cell division,
CSG is ideal and suprisingly has not been implemented in mycology agent 
based models as far as the author is aware.
That being said, a recent paper from 2023 models the time-dependence of cells of bristle 
worms (\textit{Platynereis dumerilii}) using signed distance fields (SDFs) which 
are learnt from biomedical images using neural networks, \cite{wiesner2024generative}.
As early as 2019, one could learn the continous SDF of arbitrary 3D geometry using 
\textit{DeepSDF} created by \cite{park2019deepsdf}, which employs neural networks
to learn from noisy input data. \cite{wiesner2024generative} improve 
on \textit{DeepSDF} by using \textit{sine} activation functions and 
more hidden layers in their multi-layer perceptron (MLP) neural network. Whilst 
\cite{wiesner2024generative} produce time-dependent implicit cell representations using 
an impressive data-driven approach, this thesis focuses on developing 
a mechanistic model that can bridge the gap between the cell and colony scales using 
implicit curves.
\\

The graphical technique used in the current thesis has its origin in ``Metaballs'' or 
``Blobby modelling'' which is in one case
implemented in C++ by \cite{kommareddy20143d}. The individual balls, 
modelled by quadric surfaces $f_1$ and $f_2$, are blended together 
into a new quadric surface $f(x,y,z) = f_1(x,y,z) + f_2(x,y,z)$. The idea then 
is to construct complex but smooth geometry from smaller algebraic components.
\\

\section{Algebraic topology meets mycology}

\subsection{Topologically associating domains (TADs) in genomics}

By 1997, \cite{guacci1997direct} and \cite{michaelis1997cohesins},
had independently discovered \textit{cohesins}, proteins that 
prevent ``premature splitting'' of sister chromatids due 
to microtubules, facilitating the correct timing of mitosis. 
Both teams used \textit{S. cerevisiae} cells and both
used \textit{Fluorescence In Situ Hybridization} (FISH) in order to visualise 
the budding yeast genome. Dekker et al. (2002) invented a higher 
resolution genomic imaging technique called 3C (chromosome conformation capture)
\cite{dekker2002capturing} providing a less invasive alternative 
to FISH, and using \textit{S. cerevisiae} for this comparison.
In 3C, formaldehyde fixation is used to cross-link nearby proteins.
Relative frequencies at which distinct base-pairs (from different proteins)
interact over the genome are counted by a sophisticated experimental
process known as ``ligation'' which involves a polymerase chain reaction (PCR),
\cite{dekker2002capturing}.
\\

The 3C, 4C (chromosome conformation capture-on-chip/circular chromosome conformation capture) and 5C 
(chromosome conformation capture carbon-copy) techniques have culminated in Hi-C which 
uses a combination of 3C and next generation sequencing (NGS) to image ``higher order" interactions 
in the cell nucleus, \cite{nora2012spatial}. Another pair of teams, \cite{nora2012spatial} and 
\cite{dixon2012topological}, identifed topologically associating domains (TADs) in 2012. 
They imaged mouse embryonic stem cell genomes using 5C.
\\

As indicated by the ``TAD caller'', de Wit (2019), 
weakly interacting regions of genome content that interact strongly within themselves,
called TADs (somewhat provisionally) evade clear definition, \cite{de2020tads}.
TADs are an interpretation of the diagonal blocks on a Hi-C plot, a matrix of genomic 
content proximities, \cite{de2020tads}. It is worth noting that there are 
other Hi-C structures (blocks) that have inspired new names: ``loops'' or ``compartments'', ``subTADs'',
``microTADs'', \cite{beagan2020existence}. In 2016, \cite{eser2017form} identified TADs 
in baker's yeast. The question emerges: \textit{is any subcomponent of the living (fungal) cell essential to its
growth? } De Wit (2019) suggests that \say{a more constructive path will be to explain TADs 
in light of the mechanisms that form them, rather than describing TADs as we see them}.

\subsection{A historical interlude}

\textit{A note: In this subsection I comment on 
some of the ethical dimensions of mathematical modelling in biology, 
touching on the confronting subject of eugenics. I hope that by facing 
such dehumanising content in a scientific context we 
can identify the way mathematical forms have been historically shaped by 
ideology.}
\\

Sir Ronald Alymer Fisher, whose ashes are buried in St. Peter's cathedral in Adelaide, Australia,
declared in 1959 that more ``attention to the History of Science ... by biologists is needed'',
as quoted by \cite{edwards2008gh} in an anecdotal commentary. Edwards, A (2008) cites
a selection of interesting anecdotes from Godfrey Harold Hardy and Wilhelm Weinberg who 
independently discovered the Hardy-Weinberg equilibrium and were both met by Fisher.
Fisher, who discovered the statistical technique called ANOVA, for analysis of variance, in 
\cite{fisher1919xv} (1919), quoted ``a considerable body of pedigree evidence'' for 
a ``single mendelian factor" cabable of producing ``feebleness of mind'', 
\cite{fisher1924elimination}. Fisher attempts to refute Reginald Punnett's, 
``anti-eugenic propoganda'', \cite{fisher1924elimination}, by critiquing
Punnett's data table on the number of generations required to decrease 
the proportion of ``defectives''. He subsitutes another table of 325 ``feebleminded cases'',
including ``alcoholic, sexually immoral, criminalistic, ... insane, ... tramps''.
It seems making East African \textit{pombe} is a more noble scientific cause.
\\

R. A. Fisher found creative ways to propogate his ideas. 
For example, Fisher, independently of Andrey Kolmogorov, Ivan Petrovsky, and Nikolai Piskunov,
developed a partial differential equation (PDE) known as the KPP-Fisher or
Fisher-kolmogorov equation, \cite{fisher1937wave}. At the \textit{Trinity High Table}
, Hardy is paraphrased (by Edwards, A, 2008) to have said that ``if a system of axioms 
form the deduction of a contradiction, then any proposition can be deduced from it".
The actual quote is from Fisher's 1958 anecdote \cite{edwards2008gh}, 
in which the implication that ``McTaggart and the Pope are one'', 
could be deduced if the contradiction ``two equal to one" could first be derived. 
Nowadays, new mathematical machinery has emerged which 
treat such matters with more nuance.
\\

\subsection{Grothendieck dreams}

\textit{Note: In this subsection, I name my mathematical role-model and hero, \\
Alexander Grothendieck, who I believe exemplifies the living spirit of mathematics.}
\\

Grothendieck is quoted in \cite{mclarty2007rising} to have described 
his mathematical process through anology.
\\

\say{The first analogy that came to my mind is of
immersing the nut in some softening liquid, and why not simply
water? From time to time you rub so the liquid penetrates better,
and otherwise you let time pass. The shell becomes more flexible
through weeks and monthsâ€”when the time is ripe, hand pressure
is enough, the shell opens like a perfectly ripened avocado!}
\\

The question bubbles up: \textit{Does Grothendieck's \textit{yoga} of forms, 
his pure mathematical methodology,
transfer over to applied mathematics, or even mathematical biology?}
Is not the mechanism by which \textit{cohesin} detaches from the cell's sister 
chromatids during \textit{anaphase} allowing for the splitting of 
the spindle apparatus, an example of patience and precise timing?
\\

In this literature review, I have introduced some of the technical 
and ethical dimensions of mathematical modelling in biology with a focus
on \textit{S. cerevisiae}. If topologically associated domains (TADs)
are to be taken seriously as a subject of foundational mathematical study then it 
seems conceivable that the whole spatial arrangement of a cell could 
be viewed as a TAD in itself, perhaps a ``superTAD'' to denote 
that it contains TADs, subTADs, microTADs and so on.
\\

I currently lack the expertise to formalise 
such a theory in a precise manner. That being said, 
it is interesting to consider whether the \textit{whole}
cell can be thought of as a topological space. A physical
yeast cell on a petri-dish is a messy and imperfect 
entity which appears to be the \textit{furthest} thing from 
a clean mathematical object, both in shape and function.
With the advent of tools like \textit{DeepSDF}, and topological 
data analysis (TDA), a clean mathematical description
of \textit{best fit} can be derived, even if the cell itself 
is a site of living texture. 
\\

The question of \textit{biological laws} 
then becomes: \textit{can we predict the best-mathematical-fit
for a biological assemblage, even if we cannot predict (or even 
represent completely) the actual cell?}
My hunch is that a simple ``form of best fit" really does emerge, 
and with that idea taken on a reasoned faith I proceed. Somewhat 
playfully, I supply the metaphysical interpretation that 
there is no single factor: 
\textit{Life's essence is the whole}.
\\

I represent cells as 
level-sections of polynomials in two spatial 
variables $x$, and $y$. By \textit{level-section} I mean the set of points 
$(x,y) \in \mathbb{R}^2$ that satisfy,
\begin{equation*}
    f(x,y) \geq 0,
\end{equation*}
where $f(x,y)$ is a particular member of the ring of polynomials, $\mathbb{R}[x,y]$.
In the case of an ellipse which is the polynomial, 
\begin{equation*}
    f(x,y) = 1 - \left[\left( \frac{x}{p}\right)^2 + \left( \frac{y}{q}\right)^2\right],
\end{equation*}
where $p,q \in \mathbb{R}$, there is a point of vanishing gradient at the origin. I call 
that point the center of the ellipse, which could be translated 
to new position, $(x_0,y_0)$, via the change of variables $x \mapsto x-x_0, y \mapsto y-y_0.$
In Chapter 2, the mechanism of cell division is modelled by 
combining polynomials of this form into a new one using MATLAB's \codeword{min}. 
In section \ref{sec:turingPatterns}, 
I introduce the modelling of the diffusive nutrient.


\section{The soil underneath: partial differential 
equations (PDEs) for a nutrient source} \label{sec:turingPatterns}

A reaction-diffusion PDE is employed to model the underlying 
nutrient field, $c(x,y,t)$. The yeast colony biomass given by the field, 
$b(x,y,t)$ (determined by the colony's algebraic representation) interacts 
with $c$ by a mechanism of aggregate 
gradient sensing over long periods of time \cite{keller1971model}. 
As per Keller and Segel's \textit{Model for Chemotaxis},
I represent the external force applied to the biomass, by the nutrient 
field's spatial gradient $\nabla c$, \cite{keller1971model}.
\\

The nutrient field is affected by the presence of the biomass,
and this is summarised by the reaction-diffusion PDE, \cite{fisher1937wave},
\begin{equation*}
    \pdv{c}{t} = D \left( \frac{\partial^2 c}{\partial x^2} + \frac{\partial^2 c}{\partial y^2}\right) - r b c,
\end{equation*}
where the second spatial derivatives are called the diffusion term, 
and the additional expression, $-rbc$, is a reaction between the biomass and nutrient at rate, $r$.
If $b$ is large it means the yeast is more dense so the rate of nutrient depletion 
is larger. But $c$ must be a coefficient to prevent the nutrient dropping to unphysical negative 
numbers.
\\

I finish this literature review with a wonderful insight
from the great D'Arcy Wentworth Thompson's \textit{On Growth and Form}, 
\cite{thompson1992growth},
\\

\textit{
``Still, all the while, like warp and woof, mechanism and teleology
are interwoven together, and we must not cleave to the one nor
despise the other; for their union is rooted in the very nature of
totality."}






\begin{comment}

In Keener and Sneyd's foundational textbook on cellular physiology 
(\cite{keener2009mathematical}), the law of mass action is 
introduced as the fundamental law of chemical reactions, 
the process of transformation comprises the living cell. The law of mass 
action states that for the reaction 
\begin{equation*}
    A + B \xrightarrow{k} C,
\end{equation*}
in which chemicals $A$, $B$ react to form the product $C$, the rate of change of the concentration of $C$, 
call it $c$ (in mol$\cdot L^{-1}$) is proportional to the product of the concentration 
of $A$ and $B$, given by $a$ and $b$, respectively. Sequences of interrelated chemical 
reactions induce systems of first order non-linear ordinary differential equations 
(ODEs). These are called a chemical reaction network.
\\

It is suprising then that much of the cell's functionality and behaviour 
can, in theory, be derived from these chemical reaction networks, however
we would not expect all behaviour to be derivable from this model alone.
This is for two reasons,

The art of mathematical biology is to capture a biological phenomenon with the 
simplest model possible. Another concern which appears in the connection between 
applied and pure mathematics, is that of ensuring the model that one selects is 
amenable to rigorous analysis. This further reinforces the requirement that the 
model be as simple as possible, but no simpler. Indeed, a model that is sufficiently 
crystallized so as to capture the essential details of a physical effect, will 
necessarily not capture everything observable, but has more chance to be useful, 
since it directs our thinking away from extraneous features.
\\
\\
What is essential in the study of biological systems? One essential aspect is the 
phenomenon of growth. Growth, when seen on its own, without reference to biology, 
is simply ``change over time". The standard tool for this is calculus, which in its 
more developed form, is called differential geometry. A cell colony will be modelled 
here as a subset of $\mathbb{R}^n$ where $n = 1,2,3$. In order for that set to 
``change over time", namely, for the cell to be parametrized by a real number 
$t \in \mathbb{R}_{\geq 0}$, the geometry of the colony needs to be represented 
by a finite list of parameters. The most simple way to do this, which comprises 
one of the models developed here, is to represent each cell as a disk in 
$\mathbb{R}^2$ with radius $r = r(t) \in \mathbb{R}_{\geq 0}$ centered at 
the position $(x(t),y(t)) \in \mathbb{R}^2$. The advantage to this approach 
is that it is more computationally efficient as compared to models which 
represent each cell as a deformable polygon, for instance.
\\
\\
The question remains: how can cell division be achieved within mathematical 
growth models? It does not seem obvious except via the continual addition of 
new parameters, such as new cell radii and positions. As we will see, taking 
seriously the question of mitosis from a topological standpoint will yield a 
generalized model for systems in which geometric changes and topological changes 
can be captured simultaneously. Why is the topology important in the case of biological growth? 
\\
\\
As a motivating example, consider the graph of the function $f(x) = x^2$. 
Now consider the time-dependent set given by $C(t) = f^{-1} (\{t\})$ where 
$t \in \mathbb{R}_{\geq 0}$ is a time parameter. When $t=0$, $C(0) = \{ 0\}$ 
(only one element), but when $t=1$, $C(1) = \{ -1, +1  \}$, which has two elements, 
and, likewise for $t=2$ where $C(t) = \{ -\sqrt{2}, +\sqrt{2}  \}$. This example 
shows that a catastrophic change (a bifurcation) can occur in $C(t)$ during a smooth 
change in $t$. A subtle modification to this model is to consider $f_t(x) = x^2-t$ 
and fix $C(t) = f^{-1}_t(\{0 \})$. That is, instead of sliding up the query point $t$, 
we slide down the whole smooth manifold given by $(x,f_t(x))$ such that $x \in \mathbb{R}$. 
The set $C(t)$ is actually called a level-$0$ set. We will just call this a level set of $f_t$.
\\
\\
We can consider level sets of functions from $\mathbb{R}^n$ to $\mathbb{R}$ for $n = 2,3$ 
as well. In these cases, the level sets are, respectively, level curves and level surfaces. 
A analogous example for $n=2$, is the set of functions $f_t(x,y) = x^2+y^2-t$, the level 
curves of which look like circles centered at the origin of radius $\sqrt{t}$.
\\
\\
This level of generalization will not be sufficient for our purposes, since the 
type of manifolds that we will be dealing with will not in general be representable 
as graphs of functions. To see this, consider the fact that a vertical line in the 
$xt$-plane is a perfectly reasonable representation of a stationary cell, and yet 
there's no function (of $x$) that has a vertical line graph. What is required is 
that $C(t) = \{ p \in M_t \ | \ \textrm{the last component of $p$ is $0$} \}$ where 
$M_t$ is a time dependent smooth manifold. The requirement that $M_t$ is smooth for 
all $t$ is imposed so that the theory of smooth manifolds can be brought to bear on 
the problem. What we will consider are smooth manifolds $M_t \subset \mathbb{R}^{n+1}$ 
where $n$ is the dimension in which the colony exists. For instance, a two-dimensional 
colony would be represented by
\begin{equation}
        C(t) = \{ p \in M_t \subset \mathbb{R}^3 \ | \ p_3 = 0  \}.
\end{equation}
A three-dimensional colony would be represented by
\begin{equation}
        C(t) = \{ p \in M_t \subset \mathbb{R}^4 \ | \ p_4 = 0  \}.
\end{equation}
\\
\\
In the subdiscipline of abstract algebra called ring theory, it is typical to 
consider the polynomial ring over the real field in two variables $\mathbb{R}[x,y]$ 
wherein the typical element looks like
\begin{equation}
    p(x,y) = \sum_{n,m} a_{nm} x^n y^m,
\end{equation}
where $n,m$ are non-negative integers, and $a_{nm}$ is a real coefficient.
The set $\mathbb{R}[x,y]$ is closed under multiplication and addition because 
polynomial multiplication and addition yields another real polynomial. In this 
thesis, a cell will be modeled as a level set of the quartic given by
\begin{equation}
    f(x,y) = ax^4 +bx^3+cx^2 +dx +e+
\end{equation}


At the beginning, we imagine a colony of yeast cells that is restricted to move 
along a straight line. The colony is therefore modeled as a set of real numbers 
$C \subset \mathbb{R}$. Like an archipelago of small islands, the colony as a whole 
is composed of closed sets, $C_j$ where $j \in \mathbb{N}$ indexes over the cells. 
Therefore, the colony is a disjoint union of these individual cells,
\begin{equation*}
    C = \coprod_{j=1}^N C_j
\end{equation*}
and $N$ is the total cell count. Closed sets (in the standard topology on 
$\mathbb{R}$) are chosen to represent the cells for the technical reason that 
a point (singleton) may also constitute a cell. In fact, we further restrict 
each cell to a closed interval $C_j = [a_j, b_j]$.
\\
\\
As we shall see in general, all we require is that each cell have no holes. 
Another way of saying this is that each cell is homeomorphic to a singleton 
(which works in $\mathbb{R}^2$ and $\mathbb{R}^3$ as well). Finally, the closed 
interval $[a_j,b_j]$ will be called a parametrization of the cell $C_j$: this is 
important to note for the generalization to $\mathbb{R}^2$ and $\mathbb{R}^3$, 
where parametrizations must also be constructed.
\\
\\
The mechanism of mitosis must account for the fact that several cells can 
undergo fission at the same time, or more aptly, they undergo mitosis independently. 
The most general formulation, which is also simple to implement computationally is the 
addition of a non-intersecting singleton $\{x  \}$ to the set $C$. Note, that this 
mechanism is chosen principally for the fact that it conserves the Lebesgue measure of the colony,
\begin{equation*}
    l(C \cup \{x\} ) = l(C),
\end{equation*}
where $C \cup \{x\} $ is the colony after mitosis has occurred, since singletons have measure $0$. 
\\
\\
Of course, some restrictions must be applied to the choice of 
$x \in \mathbb{R} \setminus C$. Since, we are always working in 
Euclidean spaces (for practical scenarios), we may as well require that 
$x$ is close to $C$ in the sense that for all $c \in C$ we have that 
$d(c,x) < \delta$ for some small positive amount $\delta$ where 
$d: \mathbb{R} \times \mathbb{R} \rightarrow \mathbb{R}_{\geq 0}$ 
is the Euclidean metric on $\mathbb{R}$ (which is easily generalized to 
higher dimensions). This parameter $\delta$ represents how the colony spreads out.
\\
\\
A profitable aspect of this model for cell colonies is that it unifies 
the topological aspects of the colony (it is always homeomorphic to a finite 
set of points or the empty set), and the measure theoretic properties (one can 
speak of getting more cells without changing the total measure). This means we 
can separate the dynamics of cell division from the dynamics of the geometric 
growth of the cells, i.e. the change in $a_j$ and $b_j$. One simple way to do 
this is to supply an ordinary differential equation for the measure of the 
colony $V = l(C)$, and equip this with a discrete time update equation for number of cells. 
\\
\\
To move us closer to a manageable computer implementation of cell colony growth, 
we consider the simple growth model given by,
\begin{equation*}
    \frac{dV(t)}{dt} = g(t),
\end{equation*}
\begin{equation*}
    N_{n+1} = 2N_n,
\end{equation*}
where $g(t)$ is a growth function. Now we need apply some equation for 
how each cell grows. For sake of simplicity, say that for all cells $j$, the cell measure $V_j$ grows as
\begin{equation*}
    \frac{dV_j(t)}{dt} = g_j(t), \ \ t \geq m_j,
\end{equation*}
 where $g_j(t)$ is a cellular growth function such that 
 $V_j(t) \rightarrow V_f$ is the final cellular volume, and $m_j$ is the 
 birth time of the cell. This actually is not enough to specify the whole 
 colony geometry. So how do we pin down the geometry?
\\
\\
The geometry and how its dynamics evolve of course depends on the 
chosen parametrization of each cell. For our purposes, we will 
consider $C_j = \bar{B} (x_j, \varepsilon_j)$ where
\begin{equation*}
    \bar{B} (x_j, \varepsilon_j) = \{ x \in \mathbb{R} : d(x_j,x) \leq \varepsilon_j \},
\end{equation*} 
which is called the closed ball centered on $x_j$ of radius $\varepsilon_j$.
 That means $a_j = x_j-\varepsilon_j$, and $b_j = x_j+\varepsilon_j$. It is 
 more convenient to use this parametrization since closed balls are defined 
 in higher dimensions as well. Another important benefit to the closed ball 
 is that whenever $\varepsilon_j = 0$, $\bar{B} (x_j, 0) = \{ x_j\}$. Also, 
 the volume of each cell is simply $V_j = 2\varepsilon_j$ which tells us that
  the volume is completely independent of the cell center position $x_j$.
\\
\\
Now we easily obtain a nice formula for $\varepsilon_j(t)$  defined for $t \geq m_j$ as
\begin{equation*}
    \varepsilon_j(t) = \frac{1}{2} \int_{m_j}^t g_j(s)ds.
\end{equation*}
But, recall, since the $C_j$ are each disjoint, $V$ is given by
\begin{equation*}
    V = l(C) = \sum_{j=1}^N l(C_j) = \sum_{j=1}^N V_j.
\end{equation*}
This applies to the time derivative too, yielding 
\begin{equation*}
    \frac{dV(t)}{dt} = \sum_{j=1}^N \frac{dV_j(t)}{dt}.
\end{equation*}
This means that the colony and cell growth functions must be related by the following,
\begin{equation*}
    g(t) = \sum_{j=1}^N g_j (t).
\end{equation*}
Our functions $g_j(t)$ were defined for $t \in [m_j, +\infty)$ which means they have
 different domains. To make the analysis easier, we build these functions by taking
  linear combinations of basis hat functions (defined for $t \in [0, +\infty)$). 
  The basis functions have compact support and are given piecewise by,
\begin{equation*}
    \varphi_i(t) = \begin{cases} 
      \frac{t-t_{i-1}}{h}, & t \in [t_{i-1}, t_i] \\
      \frac{t_{i+1}-t}{h}, & t \in [t_{i}, t_{i+1}] \\
      0, & \textrm{otherwise}, 
   \end{cases}
\end{equation*}
where $h$ is the smallest time step, $t_0 = 0$ and $t_i = ih$ for $i \in \mathbb{N}$.
 Hence, each $g_j(t)$ can be extended to a definition on all of $\mathbb{R}_{\geq 0}$ by
\begin{equation*}
    \tilde{g}_j(t) = \sum_{i=1}^{\infty} g_j(t_i) \varphi_i(t) =\sum_{i=1}^{\infty} g_{ij} \varphi_i(t) .
\end{equation*}
Now that the $g_{ij}$ are defined on the same domain, we can further restrict to
 $i \leq T$ where $T$ is the total number of time steps. This means that the time 
 dependence is now fixed by a finite number of parameters, $g_{ij}$. Summing over
  $j$, we get that
\begin{equation*}
    \tilde{g}(t) = \sum_{j=1}^N \sum_{i=1}^{T} g_{ij} \varphi_i(t).
\end{equation*}

\end{comment}